src/services/ViewService.js

```diff
@@ -204,6 +210,24 @@ function ViewService($q, $http, $rootScope, URLService, ElementService, UtilsSer
         return deferred.promise;
     };
 
+    var collectTableSources = function(sources, body) {
+        var i, j, k;
+        for (i = 0; i < body.length; i++) {
+            var row = body[i];
+            for (j = 0; j < row.length; j++) {
+                var cell = row[j];
+                for (k = 0; k < cell.content.length; k++) {
```

Comment:
something like underscore.js or even jQuery should have an iterable class that would allow for less error prone iteration of the table to non-recursively and with constant memory do what's being done here, right?  This looks like a loop-unrolling optimization, honestly.

Reply:
the simple for loop is usually fastest compared to other library based iterations, and we deal with a lot of huge tables, unless we need a function scope for each element, we prefer for loops for dealing with tables

Reply:
true enough.